platforms/software/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpResourceAccessor.java

```diff
@@ -16,45 +16,107 @@
 
 package org.gradle.internal.resource.transport.http;
 
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.gradle.api.resources.ResourceException;
 import org.gradle.internal.IoActions;
 import org.gradle.internal.resource.ExternalResourceName;
 import org.gradle.internal.resource.metadata.ExternalResourceMetaData;
 import org.gradle.internal.resource.transfer.AbstractExternalResourceAccessor;
 import org.gradle.internal.resource.transfer.ExternalResourceAccessor;
+import org.gradle.internal.resource.transfer.ExternalResourceReadResponse;
+import org.gradle.internal.resource.transfer.UrlExternalResource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
 import java.net.URI;
 
 public class HttpResourceAccessor extends AbstractExternalResourceAccessor implements ExternalResourceAccessor {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpResourceAccessor.class);
     private final HttpClientHelper http;
+    private final File cachePosition;
```

Comment:
I think the cache position should be determined by caller in `ExternalResourceAccessor.withContent()` rather than here.

Then the question is should I add a `File cacheLocation` inside `ExternalResourceName` or into the method  `withContent()` ? @ov7a 

I prefer add a `File cacheLocation` inside `ExternalResourceName`.

https://github.com/gradle/gradle/blob/b18f59c578ae80584df52834c2a3ba0bc9235037/platforms/software/resources/src/main/java/org/gradle/internal/resource/transfer/ExternalResourceAccessor.java#L45

Reply:
🤔 `ExternalResourceName` is an immutable ID, let's leave it like that. Adding extra method feels like an abstraction leak, but it's indeed better as far as I can see from a shallow look.

Reply:
I'll try overload the method `withContent()`:
```java
 <T> T withContent(ExternalResourceName location, boolean revalidate, @Nullable File cachePosition, ExternalResource.ContentAndMetadataAction<T> action) throws ResourceException; 
```